app/gui/view/graph-editor/src/component/edge.rs

```diff
@@ -4,2033 +4,951 @@ use crate::prelude::*;
+    /// above.
+    pub target_position: Cell<Vector2>,
+    /// Whether the edge is connected to a node input.
+    target_attached:     Cell<bool>,
+    /// Whether the edge is connected to a node output.
+    source_attached:     Cell<bool>,
+    color:               Cell<color::Rgba>,
+    /// The location of the mouse over the edge.
+    hover_position:      Cell<Option<Vector2<f32>>>,
+    disabled:            Cell<bool>,
+
+    // === Cached state ===
+    /// The endpoints of the individual [`Corner`]s making up the edge.
+    corner_points: RefCell<Vec<Vector2<f32>>>,
+
+    // === Shapes ===
+    /// The individual [`Corner`]s making up the edge. Each is drawn in the focused or unfocused
+    /// color.
+    sections:          RefCell<Vec<Rectangle>>,
+    /// A pair of [`arc`] shapes used when the mouse is over the rounded corner, and the edge must
+    /// must be split into focused and unfocused sides at a certain angle along the arc.
+    split_arc:         RefCell<Option<[arc::View; 2]>>,
+    /// Wider versions of the [`sections`], for receiving mouse events.
+    hover_sections:    RefCell<Vec<Rectangle>>,
+    /// The end of the edge that is drawn on top of the node and connects to the target node's input
+    /// port.
+    target_attachment: RefCell<Option<Rectangle>>,
+
+    // === Change detection ===
+    previous_target:          Cell<Option<Vector2<f32>>>,
+    previous_is_hoverable:    Cell<Option<bool>>,
+    previous_hover_split:     Cell<Option<EdgeSplit>>,
+    previous_color:           Cell<Option<color::Rgba>>,
+    previous_target_attached: Cell<Option<bool>>,
+}
+
+fn update_and_compare<T: Copy + PartialEq<T>, U: Into<Option<T>>>(
+    value: &Cell<Option<T>>,
+    new: U,
+) -> bool {
+    let new = new.into();
+    let old = value.replace(new);
+    old != new
+}
+
+impl EdgeModel {
+    /// Constructor.
+    #[profile(Debug)]
+    pub fn new(scene: &Scene) -> Self {
+        let display_object = display::object::Instance::new_named("Edge");
+        let source_attached = Cell::new(true);
+        let scene = scene.clone_ref();
+        let layer = &scene.layers.main_edges_level;
+        layer.add(&display_object);
+        Self {
+            display_object,
+            source_attached,
+            scene,
+            source_width: default(),
+            source_height: default(),
+            target_position: default(),
+            target_attached: default(),
+            color: default(),
+            sections: default(),
+            split_arc: default(),
+            hover_sections: default(),
+            corner_points: default(),
+            target_attachment: default(),
+            hover_position: default(),
+            disabled: default(),
+            previous_target: default(),
+            previous_is_hoverable: default(),
+            previous_hover_split: default(),
+            previous_color: default(),
+            previous_target_attached: default(),
```

Comment:
I wonder if using `..default()` is reasonable here.

Reply:
We could split this into two structs and deref to the inner struct, making the inner struct Default. I'm not sure though if we want it. Up to you :)